crates/sui-graphql-rpc/src/context_data/db_data_provider.rs

```diff
@@ -533,6 +536,33 @@ impl PgManager {
         Ok(())
     }
 
+    pub(crate) fn validate_page_limit(
+        &self,
+        first: Option<u64>,
+        last: Option<u64>,
+    ) -> Result<i64, Error> {
+        if let Some(f) = first {
+            if f > self.limits.max_page_size {
+                return Err(
+                    DbValidationError::PageSizeExceeded(f, self.limits.max_page_size).into(),
+                );
+            }
+        }
+
+        if let Some(l) = last {
+            if l > self.limits.max_page_size {
+                return Err(
+                    DbValidationError::PageSizeExceeded(l, self.limits.max_page_size).into(),
+                );
+            }
+        }
+
+        // TODO (wlmyng): even though we do not allow passing in both first and last,
+        // per the cursor connection specs, if both are provided, from the response,
+        // we need to take the first F from the left and then take the last L from the right.
+        Ok(first.or(last).unwrap_or(self.limits.max_page_size) as i64)
```

Comment:
We should not use the max page size as the default as well -- we can make the default page size a configurable limit as well, perhaps, but by setting things up this way, we're making the thing that is most expensive for us to serve the path of least resistance for users which is a bad set-up.

Reply:
Ah yeah that's fair
Any qualms if I tack it on to the service config?

Reply:
(a) No qualms!
(b) How/why are you still awake? 😬

Reply:
There are three kinds of emails that would pull me out of bed immediately

1. Correspondence from a prince about a large sum of money sitting in a bank that I have yet to claim
2. 75k points for business class flights to Seoul
2. PR reviews from @amnn

Reply:
I'm hoping that it's (2)